app/database/user_item_store_compute_all_user_items.go

```diff
@@ -93,65 +93,79 @@ func (s *UserItemStore) ComputeAllUserItems() (err error) {
-						users_items.validated = IF(task_children_data.user_item_id IS NOT NULL AND items_items.id IS NOT NULL,
+						groups_attempts.latest_activity_at = IF(children_stats.id IS NOT NULL AND items_items.id IS NOT NULL,
+							children_stats.latest_activity_at, groups_attempts.latest_activity_at),
+						groups_attempts.tasks_tried = IF(children_stats.id IS NOT NULL AND items_items.id IS NOT NULL,
+							children_stats.tasks_tried, groups_attempts.tasks_tried),
+						groups_attempts.tasks_with_help = IF(children_stats.id IS NOT NULL AND items_items.id IS NOT NULL,
+							children_stats.tasks_with_help, groups_attempts.tasks_with_help),
+						groups_attempts.tasks_solved = IF(children_stats.id IS NOT NULL AND items_items.id IS NOT NULL,
+							children_stats.tasks_solved, groups_attempts.tasks_solved),
+						groups_attempts.children_validated = IF(children_stats.id IS NOT NULL AND items_items.id IS NOT NULL,
+							children_stats.children_validated, groups_attempts.children_validated),
+						groups_attempts.validated = IF(children_stats.id IS NOT NULL AND items_items.id IS NOT NULL,
 							CASE
-								WHEN users_items.validated = 1 THEN 1
-								WHEN items.validation_type = 'Categories' THEN task_children_data.children_category = 0
-								WHEN items.validation_type = 'All' THEN task_children_data.children_non_validated = 0
-								WHEN items.validation_type = 'AllButOne' THEN task_children_data.children_non_validated < 2
-								WHEN items.validation_type = 'One' THEN task_children_data.children_validated > 0
+								WHEN groups_attempts.validated = 1 THEN 1
+								WHEN items.validation_type = 'Categories' THEN children_stats.children_non_validated_categories = 0
+								WHEN items.validation_type = 'All' THEN children_stats.children_non_validated = 0
+								WHEN items.validation_type = 'AllButOne' THEN children_stats.children_non_validated < 2
+								WHEN items.validation_type = 'One' THEN children_stats.children_validated > 0
 								ELSE 0
-							END, users_items.validated),
-						users_items.validated_at = IF(task_children_data.user_item_id IS NOT NULL AND items_items.id IS NOT NULL,
+							END, groups_attempts.validated),
+						groups_attempts.validated_at = IF(children_stats.id IS NOT NULL AND items_items.id IS NOT NULL,
 							IFNULL(
-								users_items.validated_at,
+								groups_attempts.validated_at,
 								IF(items.validation_type = 'Categories',
-									task_children_data.max_validated_at_categories, task_children_data.max_validated_at)
-							), users_items.validated_at),
-						users_items.ancestors_computation_state = 'done'
-					WHERE users_items.ancestors_computation_state = 'processing'`
+									children_stats.max_validated_at_categories, children_stats.max_validated_at)
+							), groups_attempts.validated_at),
+						groups_attempts.ancestors_computation_state = 'done'
+					WHERE groups_attempts.ancestors_computation_state = 'processing'`
```

Comment:
And related to your initial "INSERT .. ON DUPLICATE UPDATE ...", so you propagate considering the parent attempt always exists? Is it a wise assumption?

Reply:
?

Reply:
Yes, I presuppose that all the parent attempts needed for propagation are pre-created. The migration creates absent groups_attempt rows according to existing users_items.